src/main/java/com/github/pires/hazelcast/HazelcastDiscoveryController.java

```diff
@@ -119,8 +120,17 @@ public void run(String... args) {
             if (endpoints != null) {
                 if (endpoints.subsets != null && !endpoints.subsets.isEmpty()) {
                     endpoints.subsets.forEach(subset -> {
-                        subset.addresses.forEach(
+                    	if (subset.addresses != null && subset.addresses.isEmpty()) {
+                        	subset.addresses.forEach(
                                 addr -> hazelcastEndpoints.add(addr.ip));
+                        } else {
+                        	// in case of a full cluster restart
+                        	// no address might be ready, in order to allow the cluster
+                        	// to start initially, we will use the not ready addresses
+                        	// as fallback
+                        	subset.notReadyAddresses.forEach(
```

Comment:
If there are readyNodes, the notreadyaddresses are not handeled as I inserted an if/else.

Now with the new code the hazelcast will startup with no endpoints and therefore I implemented a warning message. The "issue" might be that you want a cluster but get single nodes. Otherwise with the notReady handling this shouldnt happen as long as kubernetes will not implement a third category (i.e. readyButNotHealthy or something similar)

The latest commit includes the changes you requested